src/Services/Resource.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Algolia\Settings\Services;
+
+use Algolia\Settings\Services\Exceptions\NotSearchableException;
+use AlgoliaSearch\Json;
+use Illuminate\Support\Facades\File;
+use Laravel\Scout\Searchable;
+
+final class Resource
+{
+    public function getSettings($fqn)
+    {
+        return Json::decode(
+            File::get($this->getFilePath($this->classToIndexName($fqn) . '.json')),
+            true
+        );
+    }
+
+    /**
+     * @param $fqn
+     *
+     * @return bool
+     * @throws NotSearchableException
+     */
+    public function validateClassSearchable($fqn)
+    {
+        if (! in_array(Searchable::class, class_uses_recursive($fqn))) {
+            throw new NotSearchableException(
+                'The class [' . $fqn . '] does not use the [' . Searchable::class . '] trait'
+            );
+        }
+
+        return true;
+    }
+
+    public function classToIndexName($fqn)
```

Comment:
I'd rather have a method like `getFilePath($className, $type)` that generate a absolute file name for the resource of an model. This way we can remove the public `classToIndexName`. I feel like this should stay the responsability of the Model with Searchable trait.

Reply:
And I just realized that the `classToIndex` logic is also something that exists within the narrow context of this package, especially considering the removal of the prefix code from the other PR. So it would make sense to have a single method / class responsible for getting the index name by class. The question then is how much we should obscure that from package users? Could just be omitted from the README. Could actually introduce an interface (Contract in Laravel terms... 🙄), which the Laravel docs recommend anyway, and then not put the method in the interface, which would be a strong indicator to package consumers to not rely on the function. WDYT?